### ISSUE TITLE
ATX heading incorrectly parsed as setext heading if a thematic break is immediately below

### ISSUE THREAD

**Issue by movermeyer** [yzhang-gh/vscode-markdown#879]
## Problem

Given this example Markdown file:

```markdown
## ATX Heading

Bar!

## ATX Heading with thematic break underneath

----

Foo!

## ATX Heading with thematic break directly underneath
----

Baz!

## ATX Heading with different thematic break characters directly underneath
***

Qux!

## ATX Heading with more different thematic break characters directly underneath
___

Quux!
```

I expected Markdown All-In-One to produce this Table of Contents:

```markdown
- [ATX Heading](#atx-heading)
- [ATX Heading with thematic break underneath](#atx-heading-with-thematic-break-underneath)
- [ATX Heading with thematic break directly underneath](#atx-heading-with-thematic-break-directly-underneath)
- [ATX Heading with different thematic break characters directly underneath](#atx-heading-with-different-thematic-break-characters-directly-underneath)
- [ATX Heading with more different thematic break characters directly underneath](#atx-heading-with-more-different-thematic-break-characters-directly-underneath)
```

But instead it produced:

```markdown
- [ATX Heading](#atx-heading)
- [ATX Heading with thematic break underneath](#atx-heading-with-thematic-break-underneath)
- [## ATX Heading with thematic break directly underneath](#-atx-heading-with-thematic-break-directly-underneath)
- [ATX Heading with different thematic break characters directly underneath](#atx-heading-with-different-thematic-break-characters-directly-underneath)
- [ATX Heading with more different thematic break characters directly underneath](#atx-heading-with-more-different-thematic-break-characters-directly-underneath)
```

Note that when a [thematic break](https://spec.commonmark.org/0.29/#thematic-breaks) using `-` characters is placed directly below an ATX heading, the link text in the table of contents contains the leading `#` characters of the ATX heading.

* Expected: `- [ATX Heading with thematic break directly underneath](#atx-heading-with-thematic-break-directly-underneath)`
* Produced: `- [## ATX Heading with thematic break directly underneath](#-atx-heading-with-thematic-break-directly-underneath)`

Note that using other valid thematic break characters (`*` and `_`) do not trigger this issue.

Note that the fragment in the link destination also differs between the two, but that doesn't seem to have any impact on the ability to navigate to the correct section.

# Environment

Tested with:

* Markdown All-in-One v3.4.0
* VSCode:
    * Version: 1.52.1 (user setup)
    * Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
    * Date: 2020-12-16T16:34:46.910Z
    * Electron: 9.3.5
    * Chrome: 83.0.4103.122
    * Node.js: 12.14.1
    * V8: 8.3.110.13-electron.0
    * OS: Windows_NT x64 10.0.19041

**Comment by movermeyer** [comment]
I suspect it is getting parsed as a [setext heading](https://spec.commonmark.org/0.29/#setext-headings), but it should not be:

> The lines of text must be such that, were they not followed by the setext heading underline, they would be interpreted as a paragraph: they cannot be interpretable as a code fence, ATX heading, block quote, thematic break, list item, or HTML block.
> [Setext headings](https://spec.commonmark.org/0.29/#setext-headings)

**Comment by Lemmingh** [comment]
Thanks for your contribution.
Should have been fixed in #877:

https://github.com/yzhang-gh/vscode-markdown/blob/c1e27e5c7203a7d960870147c56ebd70b8f47023/src/toc.ts#L424-L436
